springwolf-plugins/springwolf-kafka-plugin/src/main/java/io/github/stavshamir/springwolf/producer/SpringwolfKafkaProducer.java

```diff
@@ -1,72 +1,48 @@
 package io.github.stavshamir.springwolf.producer;
 
-import com.asyncapi.v2.model.server.Server;
-import com.google.common.collect.ImmutableMap;
-import io.github.stavshamir.springwolf.configuration.AsyncApiDocket;
-import lombok.Getter;
+import lombok.RequiredArgsConstructor;
 import lombok.extern.slf4j.Slf4j;
-import org.apache.kafka.clients.producer.ProducerConfig;
-import org.apache.kafka.common.serialization.StringSerializer;
-import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.kafka.core.DefaultKafkaProducerFactory;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.common.header.Header;
+import org.apache.kafka.common.header.internals.RecordHeader;
 import org.springframework.kafka.core.KafkaTemplate;
-import org.springframework.kafka.support.serializer.JsonSerializer;
-import org.springframework.stereotype.Service;
 
+import java.util.Collections;
+import java.util.List;
 import java.util.Map;
 import java.util.Optional;
+import java.util.stream.Collectors;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
 
 @Slf4j
-@Service
```

Comment:
Is this on purpose? Without this it will not be injected to the controller, or am I missing something?

Reply:
The PoducerConfiguration acts as Beanfactory and creates the Producer (but I think you have seen it already).